atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -168,6 +171,21 @@ mate::Handle<WebFrame> WebFrame::Create(v8::Isolate* isolate) {
   return mate::CreateHandle(isolate, new WebFrame(isolate));
 }
 
+v8::Local<v8::Value> WebFrame::GetResourceUsage(v8::Isolate* isolate) {
```

Comment:
You can just return `blink::WebCache::ResourceTypeStats` in this function, it will be converted automatically.